src/Types.jl

```diff
@@ -588,9 +592,16 @@ function Manifest(raw::Dict)::Manifest
         end
         deps = get(info, "deps", nothing)
         if deps !== nothing
-            for dep in deps
-                dep_uuid = UUID(raw[dep][1]["uuid"]) # check that it exists first
-                entry.deps[dep] = dep_uuid
+            if deps isa AbstractVector
+                for dep in deps
+                    dep_uuid = UUID(raw[dep][1]["uuid"]) # check that it exists first
```

Comment:
It should also check that the name is unique in the manifest file since it's invalid to use the list-of-names form when any of the names in the list are non-unique. Unless you're already verifying that elsewhere.